anime/views.py

```diff
@@ -45,31 +46,40 @@ def get_anime_by_name(name: str):
     else:
         abort(404)
 
-@animes_bp.route('/api/v1/animes/<int:id>', methods=['Put', 'Patch'])
-def anime(id: int):
+@animes_bp.route('/api/v1/animes/<int:id>', methods=['PUT', 'PATCH'])
+def update_anime(id: int):
     """Route updates attribute(s) of single anime in DB
 
-    :param name: Anime name to update its attribute(s)
-    :type name: str
+    :param id: Anime ID
+    :type ide: str
     :return: Updated anime record
     :rtype: AnimeSchema
     """
 
+    if request.method == 'PUT':
+        anime_schema = AnimeSchema()
+    else:
+        anime_schema = AnimeSchema(partial=True)
+
     try:
-        request_data = request.get_json()
-        db.session.query(Anime).filter(Anime.anime_id == id).update(
-            {**request_data}
-        )
-        db.session.commit()
+        schema_dict = anime_schema.loads(request.data)
+    except ValidationError as e:
+        return {"error": e.messages}, 400
```

Comment:
error handler

Reply:
"""
# make sure the id exists in the database

# update the data
anime_to_update = Anime(**request_data)
db.save(anime_to_update)
db.session.commit()

return 
"""

https://stackoverflow.com/questions/60545807/python-sqlalchemy-orm-update-row-with-instance-of-class

Reply:
db.save doesn't exist. I leaned to another implementation
https://stackoverflow.com/questions/60545807/python-sqlalchemy-orm-update-row-with-instance-of-class